task/task_service.go

```diff
@@ -103,7 +108,7 @@ func (ts *taskService) Create(action Action) (*Task, error) {
 		if err != nil {
 			_, markErr := ts.MarkAsFailedWithError(taskObject, err)
 			if markErr != nil {
-				fmt.Printf("[E] Task.MarkAsFailed failed: %#v\n", maskAny(markErr))
+				ts.Config.Logger.Error(nil, "[E] Task.MarkAsFailed failed: %#v", maskAny(markErr))
```

Comment:
do we want to use `context.TODO` for `nil` here? Or are we going to fix that anyway?